ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthFeeHistory.java

```diff
@@ -332,25 +345,93 @@ private List<Wei> getBaseFees(final List<BlockHeader> blockHeaders) {
         .toList();
   }
 
+  private List<Wei> getBlobBaseFees(final List<BlockHeader> blockHeaders) {
+    if (blockHeaders.isEmpty()) {
+      return Collections.emptyList();
+    }
+    // Extend block headers with their first parent
+    List<BlockHeader> extendedBlockHeaders = prependParent(blockHeaders);
+
+    // Calculate base fees for each pair of parent and headers.
+    List<Wei> baseFeesPerBlobGas =
+        IntStream.range(0, extendedBlockHeaders.size() - 1)
+            .mapToObj(
+                i -> {
+                  BlockHeader parent = extendedBlockHeaders.get(i);
+                  BlockHeader header = extendedBlockHeaders.get(i + 1);
+                  return getBlobGasFee(protocolSchedule.getByBlockHeader(header), parent);
+                })
+            .collect(Collectors.toList());
+
+    Wei nextBlobBaseFee = getNextBlobFee(blockHeaders.get(blockHeaders.size() - 1));
+    baseFeesPerBlobGas.add(nextBlobBaseFee);
+
+    return baseFeesPerBlobGas;
+  }
+
+  private Wei getNextBlobFee(final BlockHeader header) {
+    // Attempt to retrieve the next header based on the current header's number.
+    long nextBlockNumber = header.getNumber() + 1;
+    return blockchain
+        .getBlockHeader(nextBlockNumber)
+        .map(nextHeader -> getBlobGasFee(protocolSchedule.getByBlockHeader(nextHeader), header))
+        // If the next header is not present, calculate the fee using the current time.
+        .orElseGet(
+            () ->
+                getBlobGasFee(
+                    protocolSchedule.getForNextBlockHeader(header, System.currentTimeMillis()),
+                    header));
+  }
+
+  private List<BlockHeader> prependParent(final List<BlockHeader> blockHeaders) {
+    List<BlockHeader> extendedBlockHeaders = new ArrayList<>();
+    Optional<BlockHeader> parent = blockchain.getBlockHeader(blockHeaders.get(0).getParentHash());
+    parent.ifPresent(extendedBlockHeaders::add);
+    extendedBlockHeaders.addAll(blockHeaders);
+    return extendedBlockHeaders;
+  }
+
+  private Wei getBlobGasFee(final ProtocolSpec spec, final BlockHeader parent) {
+    return spec.getFeeMarket().blobGasPricePerGas(calculateExcessBlobGasForParent(spec, parent));
```

Comment:
We don't cache this on the BlockHeader like baseFee?

Reply:
nop, blobBaseFee is calculated using the parent header